include/gunrock/graph/detail/build.hxx

```diff
@@ -8,111 +8,229 @@
  *
  */
 
-#include <gunrock/graph/conversions/convert.hxx>
 #include <gunrock/algorithms/algorithms.hxx>
+#include <gunrock/formats/formats.hxx>
 
 namespace gunrock {
 namespace graph {
 namespace build {
 namespace detail {
 
 template <memory_space_t space,
-          view_t build_views,
           typename edge_t,
           typename vertex_t,
           typename weight_t>
-auto builder(vertex_t const& r,
-             vertex_t const& c,
-             edge_t const& nnz,
-             vertex_t* row_indices,
-             vertex_t* column_indices,
-             edge_t* row_offsets,
-             edge_t* column_offsets,
-             weight_t* values) {
+auto builder(graph::graph_properties_t properties,
+             format::csr_t<space, vertex_t, edge_t, weight_t>& csr) {
   // Enable the types based on the different views required.
   // Enable CSR.
-  using csr_v_t =
-      std::conditional_t<has(build_views, view_t::csr),
-                         graph::graph_csr_t<vertex_t, edge_t, weight_t>,
-                         empty_csr_t>;
-
-  // Enable CSC.
-  using csc_v_t =
-      std::conditional_t<has(build_views, view_t::csc),
-                         graph::graph_csc_t<vertex_t, edge_t, weight_t>,
-                         empty_csc_t>;
-
-  // Enable COO.
-  using coo_v_t =
-      std::conditional_t<has(build_views, view_t::coo),
-                         graph::graph_coo_t<vertex_t, edge_t, weight_t>,
-                         empty_coo_t>;
+  using csr_v_t = graph::graph_csr_t<vertex_t, edge_t, weight_t>;
+  using coo_v_t = empty_coo_t;
+  using csc_v_t = empty_csc_t;
 
-  using graph_type = graph::graph_t<space, vertex_t, edge_t, weight_t, csr_v_t,
-                                    csc_v_t, coo_v_t>;
+  using graph_type = graph::graph_t<space, vertex_t, edge_t, weight_t, csr_v_t>;
+  graph_type G(properties);
+
+  G.template set<csr_v_t>(csr.number_of_rows, csr.number_of_nonzeros,
+                          memory::raw_pointer_cast(csr.row_offsets.data()),
+                          memory::raw_pointer_cast(csr.column_indices.data()),
+                          memory::raw_pointer_cast(csr.nonzero_values.data()));
```

Comment:
Since we are embracing the vector interface, we can replace set to just take the format csr/csc/coo instead of these raw_pointer_cast and set here. And internally it can do the raw_ptrs and set them (it cannot hold structs unfortunately, because that won't work across GPU boundary). Will reduce the `builder` code a lot.

Reply:
I wanted to do this originally as well, but didn't because `set` is a `__host__ __device__` function and we need to call `.data()`. I updated this and made it a host function.

Reply:
Oh, I think we can get away without the __device__, so just a regular function.

Reply:
I thought I needed it when I was doing virtual inheritance, but it was a mess.

Reply:
Ok I got rid of __host__ too so it is just a regular function.